test/test_fx_experimental.py

```diff
@@ -1,36 +1,121 @@
 import torch
+from typing import Dict
 from torch.fx.symbolic_trace import symbolic_trace
 from torch.fx.graph_module import GraphModule
+from torch.fx.node import Node
 from torch.fx.experimental import GraphManipulation
 from torch.fx.experimental.Partitioner import Partitioner, Device, PartitionerConfig
 from torch.fx.experimental.rewriter import RewritingTracer
 from torch.testing._internal.common_utils import run_tests
 from torch.testing._internal.jit_utils import JitTestCase
-from torch.fx.experimental.partitioner_utils import NodeLatency, \
-    get_partition_to_latency_mapping, get_latency_of_partitioned_graph
+from torch.fx.experimental.partitioner_utils import (
+    NodeLatency,
+    get_partition_to_latency_mapping,
+    get_latency_of_partitioned_graph,
+)
 from typing import Union, Callable
 
+
 def symbolic_trace_with_rewrite(root: Union[torch.nn.Module, Callable]) -> GraphModule:
-    return GraphModule(root if isinstance(root, torch.nn.Module) else torch.nn.Module(), RewritingTracer().trace(root))
+    return GraphModule(
+        root if isinstance(root, torch.nn.Module) else torch.nn.Module(),
+        RewritingTracer().trace(root),
+    )
+
 
 class TestFXExperimental(JitTestCase):
+    def test_serialize_graph(self):
+        class TestModule(torch.nn.Module):
+            def __init__(self):
+                super().__init__()
+                self.linear = torch.nn.Linear(4, 4)
+                self.e = torch.rand(4)
+
+            def forward(self, a, b):
+                add_1 = a + b
+                linear = self.linear(add_1)
+                add_2 = linear + self.e
+                return add_2
+
+        m = TestModule()
+        traced = symbolic_trace(m)
+        a = torch.rand(4)
+        b = torch.rand(4)
+        GraphManipulation.get_size_of_all_nodes(traced, [a, b])
+
+        partitioner = Partitioner()
+        devices = [Device("dev_0", 5000, 0), Device("dev_1", 125, 1)]
+        partitioner_config = PartitionerConfig(devices, is_sparse_nn=True)
+        ret = partitioner.partition_graph(traced, m, partitioner_config)
+        module_with_submodules = ret.module_with_submodules
+        # Fix for now to add type/shape to output
+        for node in traced.graph.nodes:
+            if node.op == "output":
+                node.shape = a.shape
```

Comment:
Nit: Maybe we should set the shape and type of output node to be the same as `output.args[0]`? `a` might not have the correct shape or type for output node.

Reply:
I'm not too worried about the correct shape for this test. Serialization require a shape be there but it doesn't matter what the shape is.